src/main/java/org/logstash/plugins/inputs/http/NettyHttpServer.java

```diff
@@ -0,0 +1,83 @@
+package org.logstash.plugins.inputs.http;
+
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.util.concurrent.DefaultEventExecutorGroup;
+import io.netty.util.internal.SystemPropertyUtil;
+import org.logstash.plugins.inputs.http.util.SslSimpleBuilder;
+
+import java.io.Closeable;
+import java.util.concurrent.TimeUnit;
+
+import static org.logstash.plugins.inputs.http.util.DaemonThreadFactory.daemonThreadFactory;
+
+/**
+ * Created by joaoduarte on 11/10/2017.
+ */
+public class NettyHttpServer implements Runnable, Closeable {
+    private final ServerBootstrap serverBootstrap;
+    private final String host;
+    private final int port;
+
+    private static final int NUM_PROCESSORS = Runtime.getRuntime().availableProcessors();
+    static final int DEFAULT_MAX_PENDING_EXECUTOR_TASKS =
+            SystemPropertyUtil.getInt("io.netty.eventexecutor.maxPendingTasks", 16);
+
+    private final EventLoopGroup processorGroup;
+    private final DefaultEventExecutorGroup executorGroup;
+
+    public NettyHttpServer(String host, int port, IMessageHandler messageHandler,
+                           SslSimpleBuilder sslSimpleBuilder) {
+        this(host, port, messageHandler, sslSimpleBuilder, NUM_PROCESSORS);
+    }
+
+    public NettyHttpServer(String host, int port, IMessageHandler messageHandler,
+                           SslSimpleBuilder sslSimpleBuilder, int threads)
+    {
+        this.host = host;
+        this.port = port;
+        processorGroup = new NioEventLoopGroup(threads, daemonThreadFactory("http-input-processor"));
+        executorGroup = new DefaultEventExecutorGroup(threads, daemonThreadFactory("http-input-handler-executor"),
```

Comment:
minor: 
it would be swell to include the plugin id in the thread name as well.  If multiple http inputs are present in the same running instance, it is not possible from a stack track/profiler to tell which executors belong to which processors group. 

...that or just move the thread name value incrementing to static scope so that every http-input* has a unique number (and some mental math to associate them).